theano/gof/opt.py

```diff
@@ -679,7 +681,7 @@ def process_node(self, fgraph, node):
                                in zip(node_inputs_assert_removed,
                                       cand_inputs_assert_removed))
 
-            if inputs_match and node.op == candidate.op:
+            if inputs_match:
```

Comment:
Revert both changes in this file. This should be a separate PR that will need its own timing.